cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -173,39 +178,32 @@ func (p *dockerProject) Package(
 	ctx context.Context,
 	serviceConfig *ServiceConfig,
 	buildOutput *ServiceBuildResult,
-) *async.TaskWithProgress[*ServicePackageResult, ServiceProgress] {
-	return async.RunTaskWithProgress(
-		func(task *async.TaskContextWithProgress[*ServicePackageResult, ServiceProgress]) {
-			imageId := buildOutput.BuildOutputPath
-			if imageId == "" {
-				task.SetError(errors.New("missing container image id from build output"))
-				return
-			}
+) (*ServicePackageResult, error) {
```

Comment:
Implementation of methods become very simple and remove the cumbersome wrapping to support the custom task requirements.